.github/workflows/promote.yaml

```diff
@@ -0,0 +1,36 @@
+name: Promote Charm
+
+on:
+  workflow_dispatch:
+    inputs:
+      promotion:
+        type: choice
+        description: Channel to promote from
+        options:
+          - edge -> stable
+
+jobs:
+  promote:
+    name: Promote Charm
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
+      - name: Set target channel
+        env:
+          PROMOTE_FROM: ${{ github.event.inputs.promotion }}
+        run: |
+          elif [ "${PROMOTE_FROM}" == "edge -> stable" ]; then
```

Comment:
elif -> if?

Reply:
I checked this charm right now it only has 2 channels latest/edge & latest/stable

Reply:
It'd be much better if you could make the source channel configurable in case a charm is going to be promoted from the `candidate` or `beta` because for now, if the input is "beta -> stable" or "candidate -> stable", nothing would happen and the action would still succeed.
Hoever we can leave it as is if you want and improve it later